packages/babel-core/src/config/files/configuration.ts

```diff
@@ -43,30 +43,41 @@ const BABELIGNORE_FILENAME = ".babelignore";
 
 const LOADING_CONFIGS = new Set();
 
-const readConfigCode = makeStrongCache(function* readConfigCode(
+type ConfigCacheData = {
+  envName: string;
+  caller: CallerMetadata | undefined;
+};
+
+const runConfig = makeWeakCache(function* runConfig(
```

Comment:
The diff is quite big, but the main logical difference is that now this cache is initialized before calling the function rather than before importing the module, since caches must be configured synchronously.